### PR TITLE
Refactor Hilt Modules

### PR DESCRIPTION
**Pull request by jxareas**
# Changes
- Refactor Hilt modules and DI setup
- Extract functions from Mappers
- Fix project Gradle Build `ScriptCompilationException` 
- Add missing ROOM schema export directory
- Rename some files and packages by Kotlin Coding Conventions

### CODE REVIEW

File: app/src/main/java/com/uxstate/launchpad/data/mapper/Mapper.kt [hunk 1]
```diff
@@ -2,80 +2,70 @@ package com.uxstate.launchpad.data.mapper
...
 import com.uxstate.launchpad.data.local.entities.UpsEntity
-import com.uxstate.launchpad.data.remote.dto.LaunchDto
+import com.uxstate.launchpad.data.remote.dto.*
 import com.uxstate.launchpad.domain.model.*
 
-// LaunchDTO to entity
-fun LaunchDto.toPrevEntity(): PrevsEntity {
-    return PrevsEntity(
+
+fun LaunchDto.toPrevEntity(): PrevsEntity =
```
Comment (Tonnie-Dev) [inline app/src/main/java/com/uxstate/launchpad/data/mapper/Mapper.kt:9]: I like this approach using `=`

File: app/build.gradle.kts [hunk 2]
```diff
@@ -23,8 +23,13 @@ android {
         versionName = "1.0"
 
         testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
-        vectorDrawables {
-            useSupportLibrary = true
+        vectorDrawables.useSupportLibrary = true
+
+        javaCompileOptions {
+            annotationProcessorOptions {
+                arguments["room.schemaLocation"] =
```
Comment (Tonnie-Dev) [inline app/build.gradle.kts:30]: Nice Idea to add **ROOM Schema Export Directory**

File: app/schemas/com.uxstate.launchpad.data.local.LaunchDatabase/1.json [hunk 3]
```diff
@@ -0,0 +1,162 @@
+{
```
Comment (Tonnie-Dev) [inline app/schemas/com.uxstate.launchpad.data.local.LaunchDatabase/1.json:1]: I am interested in this. Do you write this manually or is there a tool to write this?

File: app/schemas/com.uxstate.launchpad.data.local.database.LaunchDatabase/1.json [hunk 4]
```diff
@@ -0,0 +1,162 @@
+{
```
Comment (Tonnie-Dev) [inline app/schemas/com.uxstate.launchpad.data.local.database.LaunchDatabase/1.json:1]: Interesting, how do you create this JSON File?

Comment (jxareas) [reply app/schemas/com.uxstate.launchpad.data.local.database.LaunchDatabase/1.json:1]: Room exports your database's schema information into a JSON file at compile time, when you set the `room.schemaLocation` annotation processor argument. This allows Room to create older versions of the database and enables automatic migration.

[For more information you can always refer to the official docs.](https://developer.android.com/training/data-storage/room/migrating-db-versions#export-schemas)

Comment (Tonnie-Dev) [reply app/schemas/com.uxstate.launchpad.data.local.database.LaunchDatabase/1.json:1]: Awesome, thanks for the tip

**Pull request merged**